examples/visual-tests/amp-inabox/amp-ads-second.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const sleep = require('sleep-promise');
+const {verifyCssElements} = require('../../../build-system/tasks/visual-diff/helpers');
+
+module.exports = {
+  'scroll down to view second ad': async (page, name) => {
+    await page.hover('#div-gpt-ad-1542810129742-0');
```

Comment:
That shouldn't really work with your current setup. Since the browser is headless, the viewport's dimensions are huge (default to 1400×100000)
You probably want to use the "viewport" setting on this test, that does two things:
1. sets the headless tab's viewport to the specific dimensions, and
2. before snapshotting and sending to Percy it wraps the entire page in an iframe of the same dimensions

That way, Percy is forced to create a snapshot of the exact dimensions you asked for, but that creates a separate issue which is that scrolling events get "lost". We hacked around this issue in the "AMP Sticky Ad" test, check out how it was built

Reply:
I will do this. Right now however the tests use iframes to house the ads' content. Using the viewport option to wrap the entire page within another iframe leads to strange behaviors to say the least. What do you think is the right way to serialize the existing iframe's contents? Modify the wrapper script to pour the content of existing iframes into the parent divs?

Reply:
I'm not sure what to do in this case... Can you give me an example of what the strange behaviours are? Do you know why they're happening?

Reply:
Looks like parsing is confused here, I'm not sure why since as you mentioned, the local browser works correctly. I'll get in touch with Percy to see if they have any idea what's going on

Reply:
This PR is blocked until the issue is fixed on percy.io, I asked them for a ~1 month timeframe. Will update once this is in. Hopefully before everyone leaves for the holidays :)

Reply:
This is resolved now.